src/asn_ber_der.ml

```diff
@@ -143,12 +145,12 @@ module R = struct
       | Constructed_indefinite when cfg.strict ->
           parse_error "Constructed indefinite form"
       | Constructed_indefinite ->
-          let (gs, tl) = children cfg eof2 [] (Cstruct.shift cs off) in
-          (G.Cons (tag, gs), Cstruct.shift tl 2)
+          let (gs, tl) = children cfg eof2 [] (String.sub cs off (String.length cs - off)) in
```

Comment:
Indeed. Do you have an application around where decoding asn1 is the bottleneck?

Reply:
is b421034 a good idea? this avoids quite some copies in the decoding process.

Reply:
I don't -- and I'm just wondering how people use this usually.  If some code paths are becoming very slow it's maybe fine, but I just have no idea of the impact :-)

Reply:
Yes, I'm concerned about performance. But so far I haven't seen an application that heavily relies on ASN.1 in their core (apart from albatross ;). It may be wise to benchmark e.g. X.509 certificates from last release and this release.